internal/pkg/scorecard/helpers.go

```diff
@@ -35,3 +40,85 @@ func ResultsCumulative(results []TestResult) (earned, max int) {
 	}
 	return earned, max
 }
+
+// CalculateResult returns a ScorecardSuiteResult with the state and Tests fields set based on a slice of ScorecardTestResults
+func CalculateResult(tests []*scapiv1alpha1.ScorecardTestResult) *scapiv1alpha1.ScorecardSuiteResult {
```

Comment:
Why are we choosing to pass in pointers and then not nil checking?

I think that the correct thing to do is not take pointers in the slice. 

I also would like to understand why a pointer back? this just means callers will have to check for nil.